crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -58,13 +58,18 @@ impl FormatNodeRule<Arguments> for FormatArguments {
                 last_node = Some(default.map_or_else(|| argument.into(), AnyNodeRef::from));
             }
 
+            // kw only args need either a `*args` ahead of them capturing all var args or a `*`
+            // pseudo-argument capturing all fields. We can also have `*args` without any kwargs
+            // afterwards.
             if let Some(vararg) = vararg {
                 joiner.entry(&format_args![
                     leading_node_comments(vararg.as_ref()),
                     text("*"),
                     vararg.format()
                 ]);
                 last_node = Some(vararg.as_any_node_ref());
+            } else if !kwonlyargs.is_empty() {
+                joiner.entry(&text("*"));
             }
```

Comment:
BTW, another place you can see some of these tricks as a sanity-check is in `Generator:

```rust
if args.vararg.is_some() || !args.kwonlyargs.is_empty() {
    self.p_delim(&mut first, ", ");
    self.p("*");
}
```